Mathlib/Topology/Algebra/Nonarchimedean/TotallyDisconnected.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2024 Jou Glasheen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jou Glasheen, Kevin Buzzard, David Loeffler, Yongle Hu, Johan Commelin
+-/
+
+import Mathlib.Topology.Algebra.Nonarchimedean.Basic
+
+/-!
+# Total separatedness of nonarchimedean groups
+
+In this file, we prove that a nonarchimedean group is a totally separated topological space.
+The fact that a nonarchimedean group is a totally disconnected topological space
+is implied by the fact that a nonarchimedean group is totally separated.
+
+## Main results
+
+- `NonarchimedeanGroup.instTotallySeparated`:
+  A nonarchimedean group is a totally separated topological space.
+
+## Notation
+
+ - `G` : Is a nonarchimedean group.
+ - `V` : Is an open subgroup which is a neighbourhood of the identity in `G`.
+
+## References
+
+See Proposition 2.3.9 and Problem 63 in [F. Q. Gouvêa, *p-adic numbers*][gouvea1997].
+-/
+
+open Pointwise TopologicalSpace
+
+variable {G : Type*} [TopologicalSpace G] [Group G] [NonarchimedeanGroup G] [T2Space G]
+
+namespace NonarchimedeanGroup
+
+@[to_additive]
+lemma exists_openSubgroup_separating {t : G} (ht : t ≠ 1) :
+    ∃ (A : Opens G) (V : OpenSubgroup G), t ∈ A ∧ 1 ∈ V ∧ Disjoint (A : Set G) V := by
```

Comment:
You can prove a more precise statement here: if `a ≠ b`, then there exists an open subgroup `V` such that `a * V` and `b * V` are disjoint.

Reply:
I am working on this.